add detailed information of the performance cluster (#2170)
kubevirt/project-infra: docs/performance-cluster.md

# Performance Cluster
Code changes may affect system behavior, scalability, and performance. So we currently have two types of tests needed to validate kubevirt code. These are:
* Correctness tests - e2e tests verifying expected system behaviors
* Performance tests - e2e tests to test system performance

Correctness tests run as unit and functional tests for each PR, and performance tests run burst density tests on a dedicated cluster in isolation to continually test builds for performance improvements and regressions. Therefore, performance testing constitutes the final layer of protection against regressions.

## Layout
![infra-layout](infra-layout.svg)

## Prow cluster
The CI/CD system for performance testing is based on [Prow], a Kubernetes CI system, using the same control plane system as the Correctness tests. Therefore, all test statuses can be viewed in the same testgrid.

> However, unlike Correctness tests, Performance testing jobs do not use KubeVirtCI, which creates a Kubernetes cluster in a virtual machine. Instead, a pre-built cluster on top of baremetal nodes is used due to performance reasons. Most speacilly because we want to avoid the performance implications of nested virtualization. Moreover, unlike Correctness tests, only one Performance test job can run at a time to avoid performance interference from workload collocation.

Reviewer: ```suggestion
However, unlike Correctness tests, Performance testing jobs do not use KubeVirtCI, which creates a Kubernetes cluster in a virtual machine. Instead, a pre-built cluster on top of baremetal nodes is used due to performance reasons. Primarily because we want to avoid the performance implications of nested virtualization. Moreover, unlike Correctness tests, only one Performance test job can run at a time to avoid performance interference from workload collocation.
```

| Day | |
| ------------- |:-------------:| 
| Mon | [100 VMs density perf test](../github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L791)  @ 04:41 AM UTC <br /> [scale VMs density perf test](../github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L878)  @ 03:41 PM UTC |
| Tue | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |
| Wed | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |
| Thu | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |
| Fri | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |
| Sat | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |
| Sun | 100 VMs density perf test  @ 04:41 AM UTC <br /> scale VMs density perf test  @ 03:41 PM UTC |

> Running a performance jobs every day would:

Reviewer: ```suggestion
Running a performance job every day would:
```

## Cluster Configuration
The cluster is a Kubernetes cluster managed on baremetal nodes in the IBM cloud. For the software layer, the cluster is configured with Kubernetes 1.21.3, where each node has CentOS Linux 8. The Kubernetes cluster consists of 3 master nodes and 3 worker nodes, the etcd cluster runs in the master nodes. 

Regarding the hardware configuration, all nodes are homogeneous, with 2 Intel(R) Xeon(R) Platinum 8260 CPUs @ 2.40GHz where each processor has 24 cores and 2 hyper-threads per core, thus with a total of 96 CPUs. In addition, each node has 128GB of DRAM, 10Gbps network bandwidth, and 1.6T of SCSI disk.

## Components
* Load generator: [perfscale-load-generator](https://github.com/kubevirt/kubevirt/tree/main/tools/perfscale-load-generator), which is a tool aimed at stressing the Kubernetes and KubeVirt control plane by creating several objects (e.g., VM, VMI, and VMIReplicaSet).

* Monitoring stack: prometheus, node-exporter, and grafana. 
   * Prometheus retains data for up to 3 months.
   * Various dashboards including a KubeVirt control-plane dashboard.

## Exposed services

> * deck: Prow UI, available at https://prow.ci.kubevirt.io

Reviewer 2: Do you consider it as helpful, if the names "periodic-kubevirt-performance-cluster-scale-density-test" and "periodic-kubevirt-performance-cluster-100-density-test" would be mentioned here again, e.g.

```suggestion
* deck: "periodic-kubevirt-performance-cluster-scale-density-test" and "periodic-kubevirt-performance-cluster-100-density-test"  in Prow UI, available at https://prow.ci.kubevirt.io
```

Author: Yes, it is a good point! Thanks

> * grafana: available at http://52.117.69.106:30000/

Reviewer 2: ```suggestion
* grafana: available at http://52.117.69.106:30000/d/V1Qq_IBM_za0/kubevirt-control-plane
```

Author: Done!

> * deck: "periodic-kubevirt-performance-cluster-100-density-test" and  "periodic-kubevirt-performance-cluster-scale-density-test" jobs in Prow UI, available at https://prow.ci.kubevirt.io

Reviewer: ```suggestion
* deck: ["periodic-kubevirt-performance-cluster-100-density-test"](https://prow.ci.kubevirt.io/?job=periodic-kubevirt-performance-cluster-100-density-test) and  ["periodic-kubevirt-performance-cluster-scale-density-test"](https://prow.ci.kubevirt.io/?job=periodic-kubevirt-performance-cluster-scale-density-test) jobs in Prow UI, available at https://prow.ci.kubevirt.io
```

* ci-search: available at https://search.ci.kubevirt.io

## Contact

> Please join our community and help us build the future of KubeVirt! There are many ways to participate. If you’re particularly interested in perfoamcne and scalability, you’ll be interested in: 

Reviewer: ```suggestion
Please join our community and help us build the future of KubeVirt! There are many ways to participate. If you’re particularly interested in performance and scalability, you’ll be interested in: 
```
